container.go

```diff
@@ -624,3 +645,64 @@ func (container *container) unregisterCallback() error {
 
 	return nil
 }
+
+// Modifies the System by sending a request to HCS
+func (container *container) Modify(config *ResourceModificationRequestResponse) error {
+	operation := "Modify"
+	title := "HCSShim::Container::" + operation
+
+	if container.handle == 0 {
+		return makeContainerError(container, operation, "", ErrAlreadyClosed)
+	}
+
+	container.handleLock.RLock()
+	defer container.handleLock.RUnlock()
+
+	requestJSON, err := json.Marshal(config)
+	if err != nil {
+		return err
+	}
+
+	var (
+		resultp *uint16
+	)
+
+	requestString := string(requestJSON)
+	logrus.Debugf(title+" id=%s request=%s", container.id, requestString)
+
+	err = hcsModifyComputeSystem(container.handle, requestString, &resultp)
+	err = processHcsResult(err, resultp)
+	if err != nil {
+		return makeContainerError(container, operation, "", err)
+	}
+	logrus.Debugf(title+" succeeded id=%s", container.id)
+	return nil
+}
+
+// ModifyContainer corresponding to the container id, by sending a request
+func ModifyContainer(id string, request *ResourceModificationRequestResponse) error {
```

Comment:
The flow remains the same for both RS1 and Current, for container create scenarios with endpoint. Both cases, OpenContainer would throw ErrComputeSystemDoesNotExist, which is treated an no error to keep the behavior same as current.

In cases when Modify is hit & Container is running (HotAdd/Remove), the RS1 Platform throw exception, which is passed on to daemon as expected. 

I'll try this  case with RS1. I may have to add a check for another Error Code "NOT_SUPPORTED" from platform. Let me check.